Source/Core/ServiceTask.swift

```diff
@@ -206,11 +209,20 @@ extension ServiceTask {
         
         metrics.fetchStartDate = Date()
         dataTask?.resume()
+        
+        // run metrics handler at end of queue
+        if !wasSuspended {
+            handlerQueue.addOperation {
+                self.sendMetrics()
+            }
+        }
+        
         return self
     }
     
     /// Suspend the underlying data task.
     public func suspend() {
+        wasSuspended = true
```

Comment:
should this get reset to `false` when `resume` is called?

Reply:
No, this property is used to track wether or not `suspend()` has ever been called. It's not meant to be a current state property, which is what the [`var state: URLSessionTask.State`](https://github.com/Electrode-iOS/ELWebService/blob/master/Source/Core/ServiceTask.swift#L27) property is for. 

Maybe there's a better var name that would clarify the intent?

Reply:
`hasEverBeenSuspended`? :)